File: backend/app/schemas/alert.py
"""Pydantic schemas for alert (risk warning) endpoints."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class AlertCreate(BaseModel):
    """Schema for creating a new alert."""

    elder_id: int
    type: str
    title: str
    description: Optional[str] = None
    risk_level: str = "low"


class AlertStatusUpdate(BaseModel):
    """Schema for updating an alert's status."""

    status: str
    remark: Optional[str] = None


class AlertBatchStatus(BaseModel):
    """Schema for batch updating alert statuses."""

    ids: list[int]
    status: str
    remark: Optional[str] = None


class AlertRecheckRequest(BaseModel):
    """Schema for triggering a rule engine recheck."""

    elder_id: int


class AlertResponse(BaseModel):
    """Schema for alert response."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    elder_id: int
    type: str
    title: str
    description: Optional[str] = None
    risk_level: str
    status: str
    source: str
    remark: Optional[str] = None
    triggered_at: Optional[datetime] = None
    created_at: datetime


class AlertListQuery(BaseModel):
    """Schema for alert list query filters."""

    page: int = 1
    page_size: int = 20
    elder_id: Optional[int] = None
    type: Optional[str] = None
    status: Optional[str] = None
    risk_level: Optional[str] = None
    date_start: Optional[str] = None
    date_end: Optional[str] = None

File: frontend/src/types/alert.ts
import type { PaginationParams } from './common';

/** Alert entity */
export interface Alert {
  id: number;
  elder_id: number;
  elder_name?: string;
  type: string;
  title: string;
  description: string;
  risk_level: 'low' | 'medium' | 'high' | 'critical';
  status: 'pending' | 'processing' | 'resolved' | 'ignored';
  source: string;
  triggered_at: string;
  resolved_at?: string;
  remark?: string;
  created_at?: string;
}

/** Create alert request */
export interface AlertCreate {
  elder_id: number;
  type: string;
  title: string;
  description: string;
  risk_level: 'low' | 'medium' | 'high' | 'critical';
}

/** Update alert status request */
export interface AlertStatusUpdate {
  status: 'pending' | 'processing' | 'resolved' | 'ignored';
  remark?: string;
}

/** Batch update alert status request */
export interface AlertBatchStatus {
  ids: number[];
  status: 'resolved' | 'ignored';
  remark?: string;
}

/** Alert list query parameters */
export interface AlertListQuery extends PaginationParams {
  elder_id?: number;
  type?: string;
  status?: string;
  risk_level?: string;
  date_start?: string;
  date_end?: string;
}
